app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
Not quite.

- remove the first `if` block (set the `<select>` value to `""` if you want)
- remove the `.tr` line and require the user to enter the name only
- set the label `<select>` value to `""`
- use something like this:

```ruby
@threads = Forumthread.none
...
label = params[:label]
label = Label.find_by(name: params[:label]) unless label == "no label"
unless label
  flash[:alert] = "'#{params[:label]}' is not a valid label."
  return
end
```